File: python/convert.py
#!/usr/bin/python3
import pandas as pd
import numpy as np
import re

data = pd.read_csv("data/movies.csv")

#print(data)

movies = pd.DataFrame(columns=['id', 'title', 'year', 'genres'])
id = 0
for index, row in data.iterrows():
  name = row['title']
  try:
    year = re.search(r'\((\d{4})\)', name).group(1)
    year = int(year)
    name = name.replace(f'({year})', '')
  except:
    year = 0
  if row['genres'] != '(no genres listed)':
    movies.loc[id] = [row['movieId'], name, year, row['genres'].split('|')]
    id+=1
  print(f'{index}/{len(data)}', end='\r')

  # limit
  #if id > 10240:
    #break
movies.to_json('data/sorted.json')
  
  

File: R/convert.r
#!/usr/bin/R -s -f
library(readxl)
library("rjson")
library("stringr")
movies <- read.csv('../python/data/movies.csv')
id = 0
movies <- na.omit(movies)
# get year out of title
movies$year = movies$title
for(i in 1:length(movies$title)){
  god = str_match(movies$title[i], '\\((\\d{4})\\)')
  movies$year[i] = as.integer(god[1, 2])
}
# sort by year
movies <- movies[order(movies$year),]
movies <- subset(movies, gsub(".*?\\((\\d{4})\\S", "\\1", movies$title) != movies$title)
# remove films with no genres
movies$genres <- gsub("\\(no genres listed)", '0', movies$genres)
movies <- subset(movies, gsub("\\(no genres listed)", '0', movies$genres) != "0")

movies$genres <- strsplit(movies$genres, split = '|', fixed=T)
# remove films with year==NA
movies <- na.omit(movies)
# save json
myfile <- toJSON(movies)
write(myfile, "myJSON.json")
